python/lsst/daf/butler/core/schema.py

```diff
@@ -19,17 +19,43 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from base64 import b64encode, b64decode
+from math import ceil
+
 from .utils import iterable, stripIfNotNone
 from .views import View
 from .config import ConfigSubset
 from sqlalchemy import Column, String, Integer, Boolean, LargeBinary, DateTime,\
-    Float, ForeignKeyConstraint, Table, MetaData
+    Float, ForeignKeyConstraint, Table, MetaData, TypeDecorator, UniqueConstraint
 
 metadata = None  # Needed to make disabled test_hsc not fail on import
 
 __all__ = ("SchemaConfig", "Schema", "SchemaBuilder")
 
 
+class Base64Bytes(TypeDecorator):
+    """A SQLAlchemy custom type that maps Python `bytes` to a base64-encoded
+    `sqlalchemy.String`.
+    """
+
+    impl = String
+
+    def __init__(self, nbytes, *args, **kwds):
+        length = 4*ceil(nbytes/3)
```

Comment:
Should it divide by 4?

Reply:
Ok, I was confused by what `nbytes` mean - I though it is stored data size. Never mind.